docs/imaging/dlc.rst

```diff
@@ -0,0 +1,114 @@
+.. _Imaging dlc:
+
+=========================
+Imaging: DLC
+=========================
+
+`DeepLabCut <http://www.mackenziemathislab.org/deeplabcut>`_ is a software package for inferring motion capture information from tracking video. Whereas Optitrack makes use of one infra-red reflectors attached to the subject(s), DeepLabCut instead uses a deep neural network trained on human-labelled data of the subject(s), allowing for less interference with subject behaviour. 
+
+
+
+Schema
+------------
+
+The DLC implementation in the imaging pipeline is shown below
+
+.. figure:: /_static/imaging/dlc_schema.png
+   :alt: DLC in the Imaging schema
+   
+   DLC in the Imaging schema
+
+* ``DLCTrackingType``
+
+  Keeps track of the kinds of tracking to which DLC can be assigned. At present, DLC data is only supported for *openfield* tasks
+
+* ``DLCModel``
+
+  Contains the full specification of a trained DLC model used in prediction. Due to the way DLC is designed, this data is derived from several separate files.
+  When creating a new Method, please pay careful attention to how body parts are named (see ``TrackedBodyPart``)
+
+* ``DLCTrackingProcessingMethod``
+
+  DLC by itself keeps track of individual body parts, not an entire subject; while the imaging pipeline is built around the assumption that a subject can be assigned to a single state vector at any given time. This table keeps track of the method to calculate that state vector given the set of body part co-ordinates provided by DLC. Note that it has to know the *exact* name of each body part in question, and so if models use new names for the same body part, a new tracking processing method must be provided. 
+
+* ``SessionDLC``
+
+  Maps a specific Sessionto a specific Processing Method
+
+* ``TrackedBodyPart``
+
+  A list of the body parts that can be tracked. Please try to avoid adding to this list unnecessarily and follow the naming schems already in use
+  
+* ``TrackingRaw.DLCPart``
+
+  Stores the raw state vector calculated via the method identifed in ``DLCTrackingProcessingMethod``
+
+* ``Tracking.DLCPart``
+
+  Stores the processed state vector, i.e. after conversion to real units, smoothing, and derivation of,e.g., speeds, angles etc. 
+
+* ``DLCPrediction``
+
+  Handles automated processing of a session
+
+
+
+Processing Methods
+-----------------------
+
+Initial work with DLC within the Moser group has focused on duplicating the same tracking methods used in the past - tracking the position and angle of the head of the subject. Typically, this took the form of tracking the left and right ears, instead of LEDs mounted to the subject's head. Consequently, exactly the same calculation method could be used, with a pre-calculation stage of identifying what data corresponds to each "LED".
+
+Future methods may be added, including potentially 3D information, but these methods must first be written.
```

Comment:
... including potentially more advanced methods to extract / triangulate tracking data (e.g. extraction of head direction from three or more points or calculations that improve robustness by improving the handling of low-likelihood tracking points, etc.)